### DISCUSSION TITLE
Modules not downloading

### DISCUSSION THREAD

**Discussion by GoofyBlaze** [cyberofficial/Synthalingua#40] [Q&A] [answered]
I was asking about this on Reddit a while ago. I ran setup a few times and this kept popping up.
![Screenshot 2023-07-27 180953](https://github.com/cyberofficial/Synthalingua/assets/139302624/a87c44e0-f02d-4e44-862e-40684eeb6695)
I looked at other discussions and it seems like I'm missing a module folder in on my end, so that may be the issue. I don't know how to solve it though.
![Screenshot 2023-07-27 180644](https://github.com/cyberofficial/Synthalingua/assets/139302624/f02cffe9-2327-4da5-8629-c023ca26c257)
Help is appreciated

**Comment by cyberofficial** [comment]
~~how are you launching the program?~~

https://github.com/cyberofficial/Synthalingua/tree/dev-testing 

I've change some minor things, could you download the dev-testing branch and see if this fixes for you? Thanks!

**Reply by GoofyBlaze** [reply]
I clicked on livetranslation twice, same goes for setup.bat
Edit: oh, had to reload trying the dev testing thing now

**Reply by GoofyBlaze** [reply]
So the module folder didn't show up, but this time it says it's missing a 'torch' module instead of 'speech_recognition' I going to try it again to see if anything changes, cause my router died during setup
![Screenshot 2023-07-27 223523](https://github.com/cyberofficial/Synthalingua/assets/139302624/f5c6ce69-69c7-44c7-92d7-353e7a351e60)

So this happened in setup

**Comment by cyberofficial** [comment]
Fixed the set up file with PR #41 

* Fixed issue with set up
* Changed the requirements file 

You can download the source here [1.0.997 (Installer Version)](https://github.com/cyberofficial/Synthalingua/releases/tag/1.0.997)
Or download the main branch through the normal method.

Please let me know if this fixes your issue. I would suggest to delete the `data_whisper` folder

**Reply by GoofyBlaze** [reply]
Alright, using the main branch the setup worked perfectly for me. I'm trying to get it to pick up japanese but I keep getting thank you over and over again.
![Screenshot 2023-07-29 023051](https://github.com/cyberofficial/Synthalingua/assets/139302624/9a65bb30-708f-41de-9d92-8b146bd55c15)
Now I tried deleting the data_whisper but that just made it go back to saying a module was missing. I tried a few different microphones but they all just gave that same thing in the image above. As far as setup goes I'm golden. I just need help with translation audio now! Your help has been amazing, I really appreciate it.

**Reply by cyberofficial** [reply]
what flag options are you using? 
In the `livetranslation.bat` file 
<img width="725" alt="explorer_q5HDPzjzQn" src="https://github.com/cyberofficial/Synthalingua/assets/19499442/092353f7-b81d-46f9-a8a2-b21efda9bd51">

the deletion of `data_whisper` was meant as reinstall, sorry if that was miscommunication.

This discussion https://github.com/cyberofficial/Synthalingua/discussions/12#discussioncomment-5499327 helped with audio, you could try some of those things out.

**Reply by GoofyBlaze** [reply]
Sorry for going silent but I checked that discussion and looked at the readme more and tried some different flag options. This is what ended up working for me right now.
![Screenshot 2023-08-07 221107](https://github.com/cyberofficial/Synthalingua/assets/139302624/cc677552-151a-4d9b-ad3e-90fed79797c3)
Does it need to use my microphone on my headset to listen and translate or do I need to try another audio device? (I see that the flag says microphone so this might be a dumb question but I just need clarification) Other than that I think I'm good. Thanks a lot!

**Reply by cyberofficial** [reply]
> Sorry for going silent but I checked that discussion and looked at the readme more and tried some different flag options. This is what ended up working for me right now. ![Screenshot 2023-08-07 221107](https://user-images.githubusercontent.com/139302624/258965668-cc677552-151a-4d9b-ad3e-90fed79797c3.png) Does it need to use my microphone on my headset to listen and translate or do I need to try another audio device? (I see that the flag says microphone so this might be a dumb question but I just need clarification) Other than that I think I'm good. Thanks a lot!

You can use `--list_microphones` flag to use other software/hardware audio outputs, most common thing is to set it to the stereo mix 
so it records from your speakers and acts like a simulated microphone.

![image](https://github.com/cyberofficial/Synthalingua/assets/19499442/cafef020-763c-4d66-8dc6-41befd57de8b) 

You can choose any device in recording section tab

If it shows here; then it'll work with the program

<img width="299" alt="ApplicationFrameHost_7EZKnG6WaY" src="https://github.com/cyberofficial/Synthalingua/assets/19499442/ce527205-6ca0-4bf5-a3f0-5587e491655f">


<img width="480" alt="explorer_NuV0ge771s" src="https://github.com/cyberofficial/Synthalingua/assets/19499442/638f32e2-0c21-4a8d-83d0-69f08f5b9779">

I use voicemeeter and use audio routing that way, but bit more complex to set up.

You can use your headset microphone to record your speakers if you want to, it's trained enough to understand not direct input so like real world conversations.

**Reply by GoofyBlaze** [reply, marked as answer]
Alright, it works! I had to get Realtek audio driver to get Stereo Mix in my sound settings and then I had to figure how to get the sound to pick up. It seems like using discussion #12 would work but I'm taking a cop out option that works just as well for me which is using earphones. I will link the 2 videos that helped me and I think we can call my problem solved. Thanks for all your help! This tool is amazing and I look forward to using it more!
Realtek Audio Install: https://youtu.be/X8ypHL-innY
Stereo Mix cop out fix: https://youtu.be/8QJydK37nIE
